lib/LedgerSMB/Report/Listings/TemplateTrans.pm

```diff
@@ -110,18 +110,14 @@ sub name {
 
 =cut
 
-my %jtype = (
-    1 => 'gl',
-    2 => 'ar',
-    3 => 'ap',
-    );
+my @jtype = ( undef, 'gl', 'ar', 'ap' );    # XXX magic number map.  From where is the magic? --rir
```

Comment:
These constants correspond with the items at https://github.com/ledgersmb/LedgerSMB/blob/master/sql/Pg-database.sql#L1197-L1202